### PR TITLE
Move browser auth to lix client

### PR DESCRIPTION
**Pull request by martin-lysk**
Moves auth to lix client and inject the required env vars to make remove the env dependency

### CODE REVIEW

File: inlang/source-code/editor/package.json
Comment (janfjohannes) [inline inlang/source-code/editor/package.json:21]: 🎈

**Review by NiklasBuchfink — APPROVED** [review]
I checked it in Fink again, LGTM ✅ 
I discovered that `manage.inlang.com` is not working at all, but that has nothing to do with this PR. I have created a issue for it: https://github.com/opral/inlang.com/issues/35

File: inlang/source-code/manage/src/components/InlangUninstall.ts
Comment (janfjohannes) [inline inlang/source-code/manage/src/components/InlangUninstall.ts:6]: dont you have to rewrite the getUser call in this file too

Comment (janfjohannes) [reply inlang/source-code/manage/src/components/InlangUninstall.ts:6]: line 173

**Review by janfjohannes — CHANGES_REQUESTED** [review]

File: lix/packages/client/src/browser-auth.ts
Comment (janfjohannes) [inline lix/packages/client/src/browser-auth.ts:157]: comment obsolete

File: lix/packages/client/src/browser-auth.ts
Comment (janfjohannes) [inline lix/packages/client/src/browser-auth.ts:1]: remove obsolete comments

File: inlang/source-code/manage/src/components/InlangUninstall.ts
Comment (martin-lysk) [reply inlang/source-code/manage/src/components/InlangUninstall.ts:6]: Good catch - how can this pass the ci - gonna fix it

Comment (martin-lysk) [reply inlang/source-code/manage/src/components/InlangUninstall.ts:6]: Just talked with @NiklasBuchfink we fix this in - we will fix the ci first https://linear.app/opral/issue/INWEB-55/update-or-deprecate-manageinlangcom and i will update the pr with the changes.

File: inlang/source-code/manage/package.json
Comment (janfjohannes) [inline inlang/source-code/manage/package.json:11]: we should not add tsc to lint and build

File: inlang/source-code/manage/tsconfig.json
Comment (janfjohannes) [inline inlang/source-code/manage/tsconfig.json:16]: remove obesolete comments

**Review by janfjohannes — CHANGES_REQUESTED** [review]

**Comment by janfjohannes** [comment]
@martin-lysk  this pr now changes the logic of lint vs build expectations officially to have tsc be part of the build instead of separate ci step or be part of lint.    before merging this, lets get everyone on board, for transparency and also just to make sure we are all aligned.

File: inlang/source-code/editor/package.json
Comment (samuelstroschein) [reply inlang/source-code/editor/package.json:10]: we had tsc as part of build scripts if I recall correctly but Ivan Hofer moved them to test scripts to speed up builds. I have no preference for either way, just adding context

Comment (janfjohannes) [reply inlang/source-code/editor/package.json:10]: the issue with test is that it runs after build and takes a long time to run and discover type issues

Comment (janfjohannes) [reply inlang/source-code/editor/package.json:10]: my favorite would be to either have it part of lint, but i would be ok to have it run before build in the build step

**Review by samuelstroschein — COMMENTED** [review]
too many changes. it's hard to navigate what i should review. 

For the sake of focusing on the memo, I neither approve nor reject the changes putting faith in you to find the right solution.

File: inlang/source-code/manage/tsconfig.json
Comment (martin-lysk) [reply inlang/source-code/manage/tsconfig.json:16]: I should have marked this pr back to draft - commited a WIP state to check ci - thanks for raising

**Comment by martin-lysk** [comment]
> too many changes. it's hard to navigate what i should review.
> 
> For the sake of focusing on the memo, I neither approve nor reject the changes putting faith in you to find the right solution.

Definitely too many changes - build pipeline should have become a separate PR. Didn't expect this one to be spread in that far :-(

File: inlang/source-code/editor/package.json
Comment (martin-lysk) [reply inlang/source-code/editor/package.json:10]: Ahh - now i see. tsc is part of test in most of the projects. We just didn't see it failing in the ci in manage and paraglide because those packages just doesn't have a test or don't have tsc in test. I would preferre to have it in build for the reason jan stated - tests are just to slow for knowing it it builds at least. I will create a separate pr for that since it changes are to hard to review now

**Pull request merged**
